# 환경 설정
---
- Anaconda Prompt에서 다음 명령 수행
> pip install opencv-python

### 버전 확인
---

In [1]:
import cv2
cv2.__version__

'4.5.5'

# OpenCV(Computer Vision)
---
- 다양한 영상 (이미지) / 동영상 처리에 사용되는 오픈소스 라이브러리

# 1. 이미지 출력
---

In [2]:
import cv2
img = cv2.imread("img.jpg") # 해당 경로의 파일 읽어오기
cv2.imshow("img", img) # img 라는 이름의 창에 img 표시
cv2.waitKey(5000) # 지정된 시간(ms) 동안 사용자가 키 입력 대기
cv2.destroyAllWindows() # 모든 창 닫기

- imshow : 제목이 img 이고 이미지 파일 이름이 img인 파일을 창에 표시한다
- waitKey : 0 = 무한 대기 , 5000(ms) = 5초
- 실행 후 q를 누르면 실행 창이 닫힌다

In [3]:
import cv2
img = cv2.imread("img.jpg") # 해당 경로의 파일 읽어오기
cv2.imshow("img", img) # img 라는 이름의 창에 img 표시
key = cv2.waitKey(5000) # 지정된 시간(ms) 동안 사용자가 키 입력 대기
print(key)
cv2.destroyAllWindows() # 모든 창 닫기

97


- key라는 변수에 key값 저장
- 아스키코드로 출력된다

## 읽기 옵션
---
1. cv2.IMREAD_COLOR : 컬러 이미지, 투명 영역은 무시(기본값)
1. cv2.IMREAD_GRAYSCALE : 흑백 이미지
1. cv2.IMREAD_UNCHANGED : 투명 영역까지 포함

In [4]:
import cv2
img_color = cv2.imread("img.jpg", cv2.IMREAD_COLOR)
img_gray = cv2.imread("img.jpg", cv2.IMREAD_GRAYSCALE)
img_unchange = cv2.imread("img.jpg", cv2.IMREAD_UNCHANGED)

cv2.imshow("img", img_color)
cv2.imshow("img_gray", img_gray)
cv2.imshow("img_unchange", img_unchange)

cv2.waitKey(0)
cv2.destroyAllWindows()

## Shape
---
- 이미지의 height, width, channel 정보

In [5]:
import cv2
img = cv2.imread("img.jpg")
img.shape # 세로 가로 Channel 정보

(390, 640, 3)

- 세로 : 390, 가로 : 640, channel : 3
- channel : 3 = RGB값을 3개 가진다.
- 투명영역에 대한 처리를 한다면 channel : 4가 된다